Sources/CacheAdvance/DataExtensions.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Created by Dan Federman on 11/10/19.
+//  Copyright © 2019 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+extension Data {
+
+    /// The width of the encoded message size.
+    static let messageSpanLength = MemoryLayout<MessageSpan>.size
+
+    /// A marker written at the end of the newest message written to disk.
+    static let endOfNewestMessageMarker = Data(MessageSpan.max)
```

Comment:
I feel like this could be one of the few places in code where it's good to pick a magic number. I think picking a distinctive number that isn't something you ever see can be valuable in cases like this. It reduces the likelihood of writing it by accident.

Speaking of that... is there any risk that someone could have a message payload that would contain just the wrong type of data that would cause us to get confused about where the end is?

If that's a possibility I wonder if there would be value in storing your metadata in a header section in the file?

Reply:
`MessageSpan.max` _is_ a magic number 🙂

> I think picking a distinctive number that isn't something you ever see can be valuable in cases like this. It reduces the likelihood of writing it by accident.

I like how you're thinking about this. The good news is I've already prevented writing it accidentally. Before you append a message, I have the following check: `messageLength <= Bytes(MessageSpan.max)`. Because I always write the message length span before the message itself, it's impossible to to accidentally write `endOfNewestMessageMarker` into the space where we write the message length span.

By picking `MessageSpan.max` as the magic value, we can clearly document what the maximum length of a message is. (It occurs to me that I could potentially use `Data(MessageSpan.min)` -- i.e. `0` -- as the delimiter. I'll investigate that separately.)

That said, I think I need to make it more clear in the code via comments how we're enforcing that we don't accidentally write the value.

> If that's a possibility I wonder if there would be value in storing your metadata in a header section in the file?

Good question! I considered doing this. In the end, it seemed simpler to write this value inline. Otherwise I'd need a separate reading/writing file handle read/write these values at the beginning (or end?) of the cache file.

Reply:
Looks like `Data(MessageSpan.zero)` works as the delimiter. I guess the question now is:

Which message is more useful to support:
1. An empty message
2. A message of extreme length

I'm leaning towards #2. Empty log messages aren't terribly useful. I think?

Reply:
Good news: `try encoder.encode("")` returns an encoded blob of length > 0. Looks like an empty message would be a weird form of data corruption.

Reply:
I like the delimiter being 0 👍